pkg/internal/util/writter_wrapper.go

```diff
@@ -0,0 +1,37 @@
+package util
+
+import "github.com/bsv-blockchain/go-sdk/util"
+
+// WrappedSdkWriter - this is temporary solution, until go-sdk and ts-sdk will agreed on handling -1 bytes as 0
+// TODO: replace with plain writer when go-sdk will be changed
+type WrappedSdkWriter struct {
+	*util.Writer
+}
+
+func NewWriter() *WrappedSdkWriter {
+	return &WrappedSdkWriter{
+		Writer: util.NewWriter(),
+	}
+}
+
+func (w *WrappedSdkWriter) WriteOptionalString(s string) {
+	if s != "" {
+		b := []byte(s)
+		w.WriteVarInt(uint64(len(b)))
+		w.WriteBytes(b)
+	} else {
+		w.writeZero()
+	}
+}
+
+func (w *WrappedSdkWriter) WriteIntBytesOptional(b []byte) {
+	if len(b) == 0 {
+		w.writeZero()
+	} else {
+		w.WriteIntBytes(b)
+	}
+}
```

Comment:
consider changing this to 
```suggestion
func (w *SdkWriter) WriteOptionalString(s string) {
	if s == "" {
		w.writeZero()
		return
	}
	w.WriteVarInt(uint64(len(s)))
	w.WriteBytes([]byte(s))
}

func (w *SdkWriter) WriteOptionalBytes(b []byte) {
	if len(b) == 0 {
		w.writeZero()
		return
	}
	w.WriteIntBytes(b)
}
```

Reply:
I like the suggestion for `WriteOptionalString` but not for `WriteOptionalBytes`.

Reply:
Besides, according to the TODO:
> TODO: replace with plain writer when go-sdk will be changed

This minor suggestion doesn't matter because it will be removed anyway - in favor of go-sdk's writer

Reply:
`WriteOptionalString` changed